core/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -99,18 +102,23 @@ public static synchronized TachyonFS get(final TachyonURI tachyonURI) throws IOE
    * @param masterHost master host details
    * @param masterPort port master listens on
    * @param zookeeperMode use zookeeper
+   * @param tachyonConf The TachyonConf instance.
    * 
-   * @return the corresponding TachyonFS hanlder
+   * @return the corresponding TachyonFS handler
    * @throws IOException
    */
-  public static synchronized TachyonFS get(String masterHost, int masterPort, boolean zookeeperMode)
-      throws IOException {
-    return new TachyonFS(new InetSocketAddress(masterHost, masterPort), zookeeperMode);
+  public static synchronized TachyonFS get(String masterHost, int masterPort,
+      boolean zookeeperMode, TachyonConf tachyonConf) throws IOException {
+    if (tachyonConf != null) {
+      tachyonConf.set(Constants.USE_ZOOKEEPER, Boolean.toString(zookeeperMode));
+      tachyonConf.set(Constants.MASTER_HOSTNAME, masterHost);
+      tachyonConf.set(Constants.MASTER_PORT, Integer.toString(masterPort));
+    }
```

Comment:
The set method could be used to modify the TachyonConf state. As for this case, yes it should have been set from client or caller.

I was trying to keep all semantic before to make the changes easy to track. I think now I could change TachyonFS#get to have 2 variance: one is the one that accept just master location, for backward compatibility, and second is the one take TachyonConf. This mean no explicit set should be done here.